cluster-autoscaler/core/scaleup/orchestrator/executor.go

```diff
@@ -31,25 +31,29 @@ import (
 	"k8s.io/autoscaler/cluster-autoscaler/context"
 	"k8s.io/autoscaler/cluster-autoscaler/metrics"
 	"k8s.io/autoscaler/cluster-autoscaler/observers/nodegroupchange"
+	"k8s.io/autoscaler/cluster-autoscaler/processors/nodegroups/asyncnodegroups"
 	"k8s.io/autoscaler/cluster-autoscaler/processors/nodegroupset"
 	"k8s.io/autoscaler/cluster-autoscaler/utils/errors"
 	"k8s.io/autoscaler/cluster-autoscaler/utils/gpu"
 )
 
 // ScaleUpExecutor scales up node groups.
 type scaleUpExecutor struct {
-	autoscalingContext *context.AutoscalingContext
-	scaleStateNotifier nodegroupchange.NodeGroupChangeObserver
+	autoscalingContext        *context.AutoscalingContext
+	scaleStateNotifier        nodegroupchange.NodeGroupChangeObserver
+	asyncNodeGroupStateCheker asyncnodegroups.AsyncNodeGroupStateChecker
 }
 
 // New returns new instance of scale up executor.
 func newScaleUpExecutor(
 	autoscalingContext *context.AutoscalingContext,
 	scaleStateNotifier nodegroupchange.NodeGroupChangeObserver,
+	asyncNodeGroupStateCheker asyncnodegroups.AsyncNodeGroupStateChecker,
```

Comment:
sorry for the late update, but I found one more typo: `asyncNodeGroupStateCheker` -> `asyncNodeGroupStateChecker`